Userland/Libraries/LibCore/ThreadEventQueue.cpp

```diff
@@ -39,15 +40,24 @@ struct ThreadEventQueue::Private {
     bool warned_promise_count { false };
 };
 
-static thread_local ThreadEventQueue* s_current_thread_event_queue;
+static pthread_key_t s_current_thread_event_queue_key;
+static pthread_once_t s_current_thread_event_queue_key_once = PTHREAD_ONCE_INIT;
 
 ThreadEventQueue& ThreadEventQueue::current()
 {
-    if (!s_current_thread_event_queue) {
-        // FIXME: Don't leak these.
-        s_current_thread_event_queue = new ThreadEventQueue;
+    pthread_once(&s_current_thread_event_queue_key_once, [] {
+        pthread_key_create(&s_current_thread_event_queue_key, [](void* value) {
+            if (value)
+                delete static_cast<ThreadEventQueue*>(value);
+        });
```

Comment:
:thinking: also, if (ptr) delete ptr is quite the anit-pattern

Reply:
cxbyte doesn't do patterns, cxbyte turns brain waves into code